storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -809,32 +809,31 @@ func newPersistentVolume(
 		)
 	}
 
-	return &corev1.PersistentVolume{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:       volume.Name,
-			Labels:     map[string]string{},
-			Finalizers: []string{VOLUME_PROTECTION},
+	pv := corev1.PersistentVolume{
+		ObjectMeta: volume.Spec.Template.Metadata,
+		Spec:       volume.Spec.Template.Spec,
+	}
+	pv.ObjectMeta.Name = volume.Name
+	pv.ObjectMeta.Finalizers = []string{VOLUME_PROTECTION}
```

Comment:
I think we may want to add a finalizer rather than overwrite them.

Reply:
yeah, good point: I'll do that.